test/jdk/sun/security/pkcs12/KeytoolOpensslInteropTest.java

```diff
@@ -23,57 +23,116 @@
+ * @run main/othervm/timeout=600 KeytoolOpensslInteropTest
  */
 
 import jdk.test.lib.Asserts;
 import jdk.test.lib.SecurityTools;
 import jdk.test.lib.process.OutputAnalyzer;
 
 import java.io.File;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
 import java.io.FileInputStream;
 import java.io.FileOutputStream;
 import java.io.IOException;
 import java.io.InputStream;
 import java.io.OutputStream;
 import java.io.UncheckedIOException;
 import java.nio.file.DirectoryStream;
-import java.nio.file.Files;
-import java.nio.file.Path;
 import java.security.KeyStore;
 import java.util.Base64;
 import java.util.Objects;
 
+import jdk.test.lib.Platform;
+import jdk.test.lib.SecurityTools;
+import jdk.test.lib.process.ProcessTools;
+import jdk.test.lib.process.OutputAnalyzer;
+import jdk.test.lib.artifacts.Artifact;
+import jdk.test.lib.artifacts.ArtifactResolver;
+import jdk.test.lib.artifacts.ArtifactResolverException;
+
 import static jdk.test.lib.security.DerUtils.*;
 import static sun.security.util.KnownOIDs.*;
 import static sun.security.pkcs.ContentInfo.*;
 
-public class ParamsTest  {
+public class KeytoolOpensslInteropTest {
 
     public static void main(String[] args) throws Throwable {
-
-        // De-BASE64 textual files in ./params to `pwd`
-        try (DirectoryStream<Path> stream = Files.newDirectoryStream(
-                Path.of(System.getProperty("test.src"), "params"),
-                p -> !p.getFileName().toString().equals("README"))) {
-            stream.forEach(p -> {
-                try (InputStream is = Files.newInputStream(p);
-                     OutputStream os = Files.newOutputStream(p.getFileName())) {
-                    Base64.getMimeDecoder().wrap(is).transferTo(os);
-                } catch (IOException e) {
-                    throw new UncheckedIOException(e);
-                }
-            });
+        String opensslPath = System.getProperty("test.openssl.path");
+        if (opensslPath == null) {
+            opensslPath = fetchOpenssl();
+        }
+        if (opensslPath != null) {
+            opensslPath = opensslPath + File.separator + "bin"
+                + File.separator + "openssl";
+            generateInitialKeystores(opensslPath);
+            testWithJavaCommands();
+            testWithOpensslCommands(opensslPath);
+        } else {
+            throw new IllegalArgumentException("Error: Can't find openssl "
+                    + "binary on this machine, please install and set openssl "
+                    + "(version 1.1.*) path with property"
+                    + " 'test.openssl.path'");
         }
+    }
+
+    private static void generateInitialKeystores(String opensslPath)
+            throws Throwable {
+        OutputAnalyzer output = ProcessTools.executeCommand(
+                opensslPath, "version")
+                .shouldHaveExitValue(0);
```

Comment:
This looks like a good time to ensure the version is 1.1.* or at least 1.* (I haven't tested with 1.0.? versions). Also, when trying out with 3.0.0, there are only 2 failures (line 119 generating weak file without -legacy. line 479 succeeds with a warning).

Reply:
ok, added version check ProcessTools.executeCommand(path, "version").shouldHaveExitValue(0).shouldMatch("1.1.*");

I tested with 1.0.2a version, test fails because some of the info messages are different.
eg: 'MAC: sha256, Iteration 10000' missing from stdout/stderr
1.0.2a  produces output as "MAC Iteration 10000" . ": sha256" is not there.